src/main/java/org/sagebionetworks/bridge/exporter/handler/SynapseExportHandler.java

```diff
@@ -190,12 +276,79 @@ private String initSynapseTable(ExportTask task) throws BridgeExporterException,
         synapseHelper.createAclWithRetry(acl);
 
         // write back to DDB table
-        Item synapseTableNewItem = new Item();
-        synapseTableNewItem.withString(getDdbTableKeyName(), getDdbTableKeyValue());
-        synapseTableNewItem.withString("tableId", synapseTableId);
-        synapseTableMap.putItem(synapseTableNewItem);
+        setSynapseTableIdToDdb(task, synapseTableId);
 
-        return synapseTableId;
+        return columnNameList;
+    }
+
+    // Helper method to get row values that are common across all Synapse tables and handlers.
+    private Map<String, String> getCommonRowValueMap(ExportSubtask subtask) {
+        ExportTask task = subtask.getParentTask();
+        Item record = subtask.getOriginalRecord();
+        String recordId = record.getString("id");
+
+        // get phone and app info
+        PhoneAppVersionInfo phoneAppVersionInfo = PhoneAppVersionInfo.fromRecord(record);
+        String appVersion = phoneAppVersionInfo.getAppVersion();
+        String phoneInfo = phoneAppVersionInfo.getPhoneInfo();
+
+        // construct row
+        Map<String, String> rowValueMap = new HashMap<>();
+        rowValueMap.put("recordId", recordId);
+        rowValueMap.put("healthCode", record.getString("healthCode"));
+        rowValueMap.put("externalId", BridgeExporterUtil.sanitizeDdbValue(record, "userExternalId", 128, recordId));
+
+        // Data groups, if present. Sort them in alphabetical order, so they appear consistently in Synapse.
+        Set<String> dataGroupSet = record.getStringSet("userDataGroups");
+        if (dataGroupSet != null) {
+            List<String> dataGroupList = new ArrayList<>();
+            dataGroupList.addAll(dataGroupSet);
+            Collections.sort(dataGroupList);
+            rowValueMap.put("dataGroups", DATA_GROUP_JOINER.join(dataGroupList));
```

Comment:
Discussed this with Alx over IM. I feel like compact code is _less_ readable, not more, so I'm not particularly inclined to make this change.